### ISSUE TITLE
support cosine metric learning？

### ISSUE THREAD

**Issue by feihuidiqiu** [ZQPei/deep_sort_pytorch#10]
It seems that the train.py does not support cosine metric learning in deepsort, is there  a plan to support that? thank you

**Comment by ZQPei** [comment]
Yeah, you can change model.py to use SOTA reid models.

**Comment by sandeshshrestha45** [comment]
Does it mean that it uses only standard softmax function ?

**Comment by dongfangduoshou123** [comment]
deep sort中位置的更新就是kalman滤波去做;
        提取的cnn特征仅用于已跟踪的目标的cnn特征
        与新检测的目标的cnn特征之间计算效率矩阵;
        效率矩阵用于确定检测与已跟踪之间的匹配问题
        （通过匈牙利算法），得到maches、unmatch_tracks、unmatch_dets分别做不同的跟踪处理，如初始化新跟踪器，update位置、mark_miss。
是这么个过程吧。

**Comment by ghm666** [comment]
> Does it mean that it uses only standard softmax function ?
I looked at the code and found that the cosine metric learning is not supported.

**Comment by ZQPei** [comment]
> > Does it mean that it uses only standard softmax function ?
> > I looked at the code and found that the cosine metric learning is not supported.

Yes, it only uses standard softmax function here, no cosine metric learning.
Pull requests is welcomed for improvement.
